php/meterpreter/meterpreter.php

```diff
@@ -512,8 +512,10 @@ function core_enumextcmd($req, &$pkt) {
 
     global $id2f;
 
-    $id_start = packet_get_tlv($req, TLV_TYPE_UINT)['value'];
-    $id_end = packet_get_tlv($req, TLV_TYPE_LENGTH)['value'] + id_start;
+    $id_start_array = packet_get_tlv($req, TLV_TYPE_UINT);
+    $id_start = $id_start_array['value'];
+    $id_end_array = packet_get_tlv($req, TLV_TYPE_LENGTH);
+    $id_end = $id_end_array['value'] + id_start;
```

Comment:
I noticed while submitting this patch that `id_start` doesn't have a `$` stuck on the front. Is this a typo? Or some PHP magic? Worth fixing?

Reply:
Looks like a bug at first glance

Reply:
Presumably it is intentionally arithmetic rather than concatenation.

A few tests on PHP 7:

```
# php --version
PHP 7.4.5 (cli) (built: Jun  3 2020 20:55:00) ( NTS )
Copyright (c) The PHP Group
Zend Engine v3.4.0, Copyright (c) Zend Technologies
    with Zend OPcache v7.4.5, Copyright (c), by Zend Technologies

# php -r '$asdf = 123; echo "hello" + asdf;'
PHP Warning:  Use of undefined constant asdf - assumed 'asdf' (this will throw an Error in a future version of PHP) in Command line code on line 1
PHP Warning:  A non-numeric value encountered in Command line code on line 1
PHP Warning:  A non-numeric value encountered in Command line code on line 1
0

# php -r '$asdf = "123"; echo "hello" + asdf;'
PHP Warning:  Use of undefined constant asdf - assumed 'asdf' (this will throw an Error in a future version of PHP) in Command line code on line 1
PHP Warning:  A non-numeric value encountered in Command line code on line 1
PHP Warning:  A non-numeric value encountered in Command line code on line 1
0

# php -r '$asdf = 123; echo "hello" + $asdf;'
PHP Warning:  A non-numeric value encountered in Command line code on line 1
123

# php -r '$asdf = "123"; echo "hello" + $asdf;'
PHP Warning:  A non-numeric value encountered in Command line code on line 1
123

# php -r '$asdf = 123; echo 999 + $asdf;'
1122

# php -r '$asdf = 123; echo 999 + asdf;'
PHP Warning:  Use of undefined constant asdf - assumed 'asdf' (this will throw an Error in a future version of PHP) in Command line code on line 1
PHP Warning:  A non-numeric value encountered in Command line code on line 1
999
```

Reply:
Hmm

If I add to Meterpreter:

```php
echo "\n";
echo gettype($id_start) . ": $id_start\n";
echo gettype($id_end) . ": $id_end\n";

```

Then without the `$` in front of `id_start` I get:

```
$ php pwn.php 
/*
integer: 1000
integer: 1000
```

With `$id_start` I get:

```
$ php pwn.php 
/*
integer: 1000
integer: 2000
```

Looking at fc0079e98 (especially the Python code) it looks like the intention was the latter.

But the Python code seems to have its own error (`start` should be `id_start`, right?)

Reply:
OJ confirmed the missing `$` was a typo. I'll fix.

Reply:
> Then without the `$` in front of `id_start` I get:
> 
> ```
> $ php pwn.php 
> /*
> integer: 1000
> integer: 1000
> ```
> 
> With `$id_start` I get:
> 
> ```
> $ php pwn.php 
> /*
> integer: 1000
> integer: 2000
> ```
> 
> Looking at [fc0079e](https://github.com/rapid7/metasploit-payloads/commit/fc0079e98a6a4b5f1ce26d8e455a39c3ad34bc24) (especially the Python code) it looks like the intention was the latter.

Agreed. Without bothering to do any tests, it appears that the intention was to perform addition, in which case the second example is correct. That is `$start_id`, with `$`.